src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public class zip_LargeFiles : ZipFileTestBase
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+        [OuterLoop("It requires almost 12 GB of free disk space")]
+        public static void UnzipOver4GBZipFile()
+        {
+            byte[] buffer = new byte[1_000_000_000]; // 1 GB
+
+            string zipArchivePath = Path.Combine("ZipTestData", "over4GB.zip");
+            DirectoryInfo tempDir = Directory.CreateDirectory(Path.Combine("ZipTestData", "over4GB"));
+
+            try
+            {
+                for (byte i = 0; i < 6; i++)
+                {
+                    buffer.AsSpan().Fill(i);
+
+                    string entryName = $"{i}.test";
+                    File.WriteAllBytes(Path.Combine(tempDir.FullName, entryName), buffer);
+                }
+
+                ZipFile.CreateFromDirectory(tempDir.FullName, zipArchivePath, CompressionLevel.NoCompression, includeBaseDirectory: false);
+
+                using ZipArchive zipArchive = ZipFile.OpenRead(zipArchivePath);
+                foreach (ZipArchiveEntry entry in zipArchive.Entries)
+                {
+                    using Stream entryStream = entry.Open();
+
+                    Assert.True(entryStream.CanRead);
+                    Assert.Equal(buffer.Length, entryStream.Length);
+                }
+            }
+            finally
+            {
+                if (File.Exists(zipArchivePath))
```

Comment:
does'nt matter much, but I think File.Delete() succeeds if the file doesn't exist.

Reply:
@danmoseley you are right (I was not aware of that, but I've verified that)

Reply:
similar to Directory.CreateDirectory does not need a check it doesn't exist.